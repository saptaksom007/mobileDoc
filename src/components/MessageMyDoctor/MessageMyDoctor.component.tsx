import { connect } from 'react-redux'
import { Icon } from 'react-native-elements'
import { Paragraph } from 'components/CustomText/CustomText.component'
import { View, StyleProp, ViewStyle } from 'react-native'
import { Color } from 'constants/Color'
import ConversationsListActions from 'screens/ConversationsList/ConversationsList.actions'
import DashboardActions from 'screens/Dashboard/Dashboard.actions'
import i18n from 'ex-react-native-i18n'
import primaryConSelector from 'common-docdok/lib/domain/messaging/selectors/primaryConSelector'
import { findPatientSelf } from 'common-docdok/lib/domain/healthrelation/selectors/findPatient'
import { hasOnlyOneConversation } from 'common-docdok/lib/domain/messaging/selectors/findConversation'

import React, { PureComponent } from 'react'
import { Actions as NavAction } from 'navigation/SagaNavigation'
import TouchableNativeFeedbackSafe from '@expo/react-native-touchable-native-feedback-safe'

import { getTestId } from 'utilities/environment'

import styles from './MessageMyDoctor.styles'
import { Action } from 'redux'
import { UserOrigins } from 'constants/UserOrigins'

const testIdMyDoctor = getTestId('MessageMyDoctor')

interface Props {
  primaryConversation?: any
  dispatch?(action: Action): void
  isLoggedIn?: boolean
  onlyOneConversation?: boolean
  forceConversationId?: number
  conversations?: { [id: number]: any }
  containerStyle?: StyleProp<ViewStyle>
  otherFilter(conversation: any): void
  origin?: string
}

export class MessageMyDoctor extends PureComponent<Props> {
  static defaultProps = {
    containerStyle: {},
    otherFilter: () => true,
  }

  nextRoute = (route: any) => {
    const { dispatch, isLoggedIn } = this.props
    if (isLoggedIn) {
      dispatch!(route)
    } else {
      dispatch!(DashboardActions.loginWithNextRouteRequested(route))
    }
  }

  gotoChat() {
    const {
      forceConversationId,
      otherFilter,
      onlyOneConversation,
      primaryConversation,
    } = this.props
    if (forceConversationId) {
      this.nextRoute(
        ConversationsListActions.gotoChatRequested(forceConversationId),
      )
    } else if (onlyOneConversation && primaryConversation) {
      this.nextRoute(
        ConversationsListActions.gotoChatRequested(primaryConversation.id),
      )
    } else {
      this.nextRoute(NavAction.push('conversationslist', { otherFilter }))
    }
  }

  render() {
    const { isLoggedIn, containerStyle, origin } = this.props
    const isMigros = false && origin === UserOrigins.MIGROS

    if (!isLoggedIn) {
      return null
    }
    return (
      <View
        style={[styles.container, containerStyle]}
        testID={testIdMyDoctor}
        accessibilityLabel={testIdMyDoctor}
      >
        <View style={styles.textContainer}>
          <Paragraph style={styles.messageMyDoctorText} numberOfLines={2}>
            {i18n.t(
              isMigros
                ? `dashboard.messageMyDoctorMIGROS`
                : 'dashboard.messageMyDoctor',
            )}
          </Paragraph>
        </View>
        <TouchableNativeFeedbackSafe
          onPress={() => this.gotoChat()}
          activeOpacity={0.7}
        >
          <Icon name={'email'} size={55} color={Color.tintColor} />
        </TouchableNativeFeedbackSafe>
      </View>
    )
  }
}

function select(state: any): any {
  const {
    login: { isLoggedIn },
    healthrelation: { patients },
    messaging: { conversations },
  } = state

  const patient = patients.find(findPatientSelf)
  const primaryConversation =
    patient && conversations[primaryConSelector(state, patient.uuid)!]

  const onlyOneConversation = hasOnlyOneConversation(
    Object.values(conversations),
  )

  return {
    primaryConversation,
    isLoggedIn,
    patientSelf: !!patient,
    conversations,
    onlyOneConversation,
  }
}

export default connect(select)(MessageMyDoctor)
