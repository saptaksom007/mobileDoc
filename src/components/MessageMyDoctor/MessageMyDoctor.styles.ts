import { StyleSheet } from 'react-native'
import { Layout } from 'constants/Layout'

export const ICON_SIZE_LARGE: number = Layout.window.height * 0.09
export const ICON_SIZE_SMALL: number = Layout.window.height * 0.03

export default StyleSheet.create({
  container: {
    width: Layout.window.width,
    height: ICON_SIZE_LARGE + 25,
    marginTop: 17,
    alignItems: 'center',
    justifyContent: 'center'
  },
  textContainer: {
    justifyContent: 'flex-end',
    backgroundColor: 'transparent',
    height: 40
  },
  messageMyDoctorText: {
    textAlign: 'center'
  }
})
