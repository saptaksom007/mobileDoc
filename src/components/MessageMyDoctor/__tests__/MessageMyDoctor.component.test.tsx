import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { MessageMyDoctor } from 'components/MessageMyDoctor/MessageMyDoctor.component'

it('<MessageMyDoctor/> log out', () => {
  const tree = renderer.create(<MessageMyDoctor />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('<MessageMyDoctor/> log in', () => {
  const tree = renderer.create(<MessageMyDoctor isLoggedIn />).toJSON()
  expect(tree).toMatchSnapshot()
})
