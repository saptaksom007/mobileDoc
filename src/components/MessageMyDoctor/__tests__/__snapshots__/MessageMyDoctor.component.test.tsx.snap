// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageMyDoctor/> log in 1`] = `
<View
  accessibilityLabel="MessageMyDoctor"
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 145.06,
        "justifyContent": "center",
        "marginTop": 17,
        "width": 750,
      },
      Object {},
    ]
  }
  testID="MessageMyDoctor"
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 40,
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      allowFontScaling={false}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "TitilliumWeb-Regular",
            "fontSize": 17.5,
            "lineHeight": 22,
            "overflow": "visible",
          },
          Object {
            "color": "rgba(27, 106, 158, 0.85)",
            "fontSize": 17.5,
            "lineHeight": 21,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      dashboard.messageMyDoctor
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View />
  </View>
</View>
`;

exports[`<MessageMyDoctor/> log out 1`] = `null`;
